### PR TITLE
Navigation: Use navId and pageNav for Create User page

### LINKED ISSUE #55107: Page layout: Server admin -> Users -> New user
- [ ] convert to use `navId` instead of `navModel`
- [ ] add `subTitle`
- [ ] add `pageNav` (see `TeamPages`)

### PR DESCRIPTION
**Pull request by lpskdl**
**What this PR does / why we need it**:
-  convert to use navId instead of navModel
-  add subTitle
-  add pageNav

**topnav on**
Screenshot 2022-09-14 at 23 24 41

**topnav off**
Screenshot 2022-09-14 at 23 57 17

**Which issue(s) this PR fixes**:

Fixes #55107 

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/admin/UserCreatePage.tsx [hunk 1]
```diff
@@ -22,7 +15,15 @@ interface UserDTO {
 
 const createUser = async (user: UserDTO) => getBackendSrv().post('/api/admin/users', user);
 
-const UserCreatePage: React.FC<UserCreatePageProps> = ({ navModel }) => {
+const pageNav: NavModelItem = {
+  icon: 'user',
+  id: 'team-new',
+  text: 'Add new user',
+  subTitle: 'Add new user information.',
```
Comment (torkelo) [inline public/app/features/admin/UserCreatePage.tsx:22]:
```suggestion
  subTitle: 'Create a new Grafana user.',
```

**Review by torkelo — APPROVED** [review]
While you are changing this file can you change so the user get's redirected to the user page after create?

Change this code so it redirects to the user page (using id from the data userDTO) 
  const onSubmit = useCallback(
    async (data: UserDTO) => {
      await createUser(data);
      history.push('/admin/users');
    },
    [history]
  );

File: public/app/features/admin/UserCreatePage.tsx [hunk 2]
```diff
@@ -22,21 +15,29 @@ interface UserDTO {
 
 const createUser = async (user: UserDTO) => getBackendSrv().post('/api/admin/users', user);
 
-const UserCreatePage: React.FC<UserCreatePageProps> = ({ navModel }) => {
+const pageNav: NavModelItem = {
+  icon: 'user',
+  id: 'team-new',
+  text: 'Add new user',
```
Comment (ashharrison90) [inline public/app/features/admin/UserCreatePage.tsx:21]: i would just make this `New user` tbh

```suggestion
  text: 'New user',
```

**Review by ashharrison90 — APPROVED** [review]
looks great, minor nit on the title 🙌

**Pull request merged**